vendor/src/github.com/docker/distribution/registry/api/v2/names.go

```diff
@@ -18,7 +18,7 @@ const (
 // RepositoryNameComponentRegexp restricts registry path component names to
 // start with at least one letter or number, with following parts able to
 // be separated by one period, dash or underscore.
-var RepositoryNameComponentRegexp = regexp.MustCompile(`[a-z0-9]+(?:[._-][a-z0-9]+)*`)
+var RepositoryNameComponentRegexp = regexp.MustCompile(`[a-z0-9]+(?:[._-]+[a-z0-9]+)*`)
```

Comment:
Actually no. Regex can't be simplified to that. There is one very slight difference. The regex before ensures that an alnum must follow the `._-`. New one doesn't.